.github/workflows/test.yml

```diff
@@ -339,10 +402,20 @@ jobs:
       with:
         ruby-version: 2.6
     - run: bundle lock
-    - uses: actions/cache@v1
+    - uses: actions/cache@v2
+      name: cache pip dependencies
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
```

Comment:
same here - this might need to include `${{ matrix.python }}` as part of the `key` and restore key